File: phase_Two/leetcode/longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        i = 0
        j = 0  
        k = 1  
        mx = 0
        
        while j < len(nums):
            if nums[j] == 0:
                if k > 0:
                    k -= 1
                    j += 1
                elif k <= 0:
                    k = k + 1 if nums[i] == 0 else k
                    i += 1
            else:
                j += 1
            mx = max(mx, j - i)
        
        return mx - 1

File: phase_Two/leetcode/longest-subarray-of-1s-after-deleting-one-element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
          int i = 0;
        int j = 0;  
        int k =1 ; 
        int mx = 0;
        while(j < nums.size())
        {
            if(nums[j]==0){
              if(k>0){
              
                k--;
                j++;
                
              }
              else if(k<=0){
                k = (nums[i]==0)?k+1 : k ;
              i++;
              }
             
            }
            else{
              j++;
            }
           mx = max(mx,j-i);
        }
        return mx-1;
    }
};